core/common/src/main/java/alluxio/resource/ResourcePool.java

```diff
@@ -88,7 +112,17 @@ public T acquire() {
           if (resource != null) {
             return resource;
           }
-          mNotEmpty.await();
+          if (time != null) {
+            long currTimeMs = System.currentTimeMillis();
```

Comment:
I think this part needs to be handled differently, ie. the time that is passed in should accurately reflect how much time was spent in `mNotEmpty.await` and not other portions of the code.